### ISSUE TITLE
Cannot read properties of undefined (reading 'spanId')

### ISSUE THREAD

**Issue by zhenyuT** [openzipkin/zipkin#3521]
When upgrade zipkin from 2.14 to latest(2.24),I get the following error when clicking the SHOW button on certain traces.

```
TypeError: Cannot read properties of undefined (reading 'spanId')
    at ao (Timeline.tsx:88:50)
    at Xo (react-dom.production.min.js:153:146)
    at ms (react-dom.production.min.js:261:496)
    at cl (react-dom.production.min.js:246:265)
    at sl (react-dom.production.min.js:246:194)
    at $s (react-dom.production.min.js:239:172)
    at react-dom.production.min.js:123:115
    at t.unstable_runWithPriority (scheduler.production.min.js:19:467)
    at Gi (react-dom.production.min.js:122:325)
    at Yi (react-dom.production.min.js:123:61)
```
![image](https://user-images.githubusercontent.com/9477681/230878313-4a50b677-8c0a-4ad5-94c5-7f572b025a96.png)

I've tried to debug the code.The reason for the problem may be due to the lack of proper error handling.

![image](https://user-images.githubusercontent.com/9477681/230880602-e05492e3-823a-4ce7-abbf-0b3be529afac.png)

![image](https://user-images.githubusercontent.com/9477681/230895692-33af0e07-5b87-4904-af0b-9b6542d27a67.png)

src/components/TracePage/helpers.ts
line 32
```
const roots = spans.filter((span) => !span.parentId);
```
The roots that returned is an empty array.
And I  check my data,it just do not have th expected span.
But it works well on the old UI.

**Comment by jcchavezs** [comment]
Ping @tacigar

**Comment by zhenyuT** [comment]
Do I need to upload the trace data for testing?

**Comment by p1742000** [comment]
is this issue resolved ? We are also facing same issue on this version zipkin-lens 2.24.0.

**Comment by wu-sheng** [comment]
Add a hook for https://github.com/apache/skywalking/discussions/10877#discussioncomment-6089542, as this is asked on SkyWalking as well.

**Comment by tacigar** [comment]
This issue was resolved by https://github.com/openzipkin/zipkin/pull/3507
And the patch is included by 2.24.1 🙇

**Comment by p1742000** [comment]
> This issue was resolved by #3507 And the patch is included by 2.24.1 🙇

We used zipkin version 2.24.1 only, but still in the UI same issue is there and in the console same error is there.

<img width="619" alt="image" src="https://github.com/openzipkin/zipkin/assets/61838673/f680ffbc-4ee7-4bb7-ab7a-b60d68b49c07">


One of the trace details is here :

"traceId":"2626155b696501ad","parentId":"1c3f553277536de9","id":"8229274fc6e56c9a","kind":"SERVER","name":"ratings.default.svc.cluster.local:9080/*","timestamp":1685943963029759,"duration":1852,"localEndpoint":{"serviceName":"ratings.default","ipv4":"10.1.2.70"},"annotations":[{"timestamp":1685943963031747,"value":"ss"}],"tags":{"component":"proxy","downstream_cluster":"-","guid:x-request-id":"c46f6948-8804-467c-a7e6-4f0c7db8e6b6","http.method":"GET","http.protocol":"HTTP/1.1","http.status_code":"200","http.url":"[http://ratings:9080/ratings/0","istio.canonical_revision":"v1","istio.canonical_service":"ratings","istio.mesh_id":"cluster.local","istio.namespace":"default","node_id":"sidecar10.1.2.70ratings-v1-5f9699cfdf-8t8b8.default~default.svc.cluster.local","peer.address":"10.1.2.74","request_size":"0","response_flags":"-","response_size":"48","upstream_cluster":"inbound|9080||","upstream_cluster.name":"inbound|9080||","user_agent":"Mozilla/5.0](http://ratings:9080/ratings/0%22,%22istio.canonical_revision%22:%22v1%22,%22istio.canonical_service%22:%22ratings%22,%22istio.mesh_id%22:%22cluster.local%22,%22istio.namespace%22:%22default%22,%22node_id%22:%22sidecar~10.1.2.70~ratings-v1-5f9699cfdf-8t8b8.default~default.svc.cluster.local%22,%22peer.address%22:%2210.1.2.74%22,%22request_size%22:%220%22,%22response_flags%22:%22-%22,%22response_size%22:%2248%22,%22upstream_cluster%22:%22inbound%7C9080%7C%7C%22,%22upstream_cluster.name%22:%22inbound%7C9080%7C%7C%22,%22user_agent%22:%22Mozilla/5.0) (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36"




trace data is normal, then also why are we facing this issue. are we missing Anything ?  @tacigar
